backend/src/sms/middlewares/sms-template.middleware.ts

```diff
@@ -63,6 +64,40 @@ const storeTemplate = async (req: Request, res: Response, next: NextFunction): P
     return next(err)
   }
 }
+
+/**
+ * Updates the campaign and email_messages table in a transaction, rolling back when either fails.
+ * For campaign table, the s3 meta data is updated with the uploaded file, and its validity is set to true.
+ * For email_messages table, existing records are deleted and new ones are bulk inserted.
+ * @param key 
+ * @param campaignId
+ * @param filename 
+ * @param records
+ */
+const updateCampaignAndMessages = async (
+  key: string,
+  campaignId: string,
+  filename: string,
+  records: MessageBulkInsertInterface[]): Promise<void> => {
+  let transaction
+
+  try {
+    transaction = await Campaign.sequelize?.transaction()
+    // Updates metadata in project
+    await CampaignService.updateCampaignS3Metadata( key, campaignId, filename, transaction)
+
+    // START populate template
+    await SmsTemplateService.addToMessageLogs(+campaignId, records, transaction)
+
+    // Set campaign to valid
+    await CampaignService.setValid(+campaignId, transaction)
```

Comment:
This was initially inside `addToMessageLogs`, moved it out to make the code more readable.